agent-install/agent-install.sh

```diff
@@ -515,207 +586,349 @@ function install_linux(){
-            set -x
-            hzn register -m "${NODE_NAME}" -o "$HZN_ORG_ID" -u "$HZN_EXCHANGE_USER_AUTH" -n "$HZN_EXCHANGE_NODE_AUTH"
-            set +x
+        	if [[ -z "${3}" ]]; then
+        		log_info "Neither a pattern nor node policy were not specified, registering without it..."
+            	set -x
+            	hzn register -m "${NODE_NAME}" -o "$HZN_ORG_ID" -u "$HZN_EXCHANGE_USER_AUTH" -n "$HZN_EXCHANGE_NODE_AUTH"
+            	set +x
+        	else
+        		log_info "Node policy ${HZN_NODE_POLICY} was specified, registering..."
+            	set -x
+            	hzn register -m "${NODE_NAME}" -o "$HZN_ORG_ID" -u "$HZN_EXCHANGE_USER_AUTH" -n "$HZN_EXCHANGE_NODE_AUTH" --policy "$3"
+            	set +x
+        	fi
         else
-            echo `now` "Registering node with ${2} pattern"
-            set -x
-            hzn register -p "$2" -m "${NODE_NAME}" -o "$HZN_ORG_ID" -u "$HZN_EXCHANGE_USER_AUTH" -n "$HZN_EXCHANGE_NODE_AUTH"
-            set +x
+        	if [[ -z "${3}" ]]; then
+        		log_info "Registering node with ${2} pattern"
+            	set -x
+            	hzn register -p "$2" -m "${NODE_NAME}" -o "$HZN_ORG_ID" -u "$HZN_EXCHANGE_USER_AUTH" -n "$HZN_EXCHANGE_NODE_AUTH"
+            	set +x
+        	else
+        		log_info "Pattern ${2} and policy ${3} were specified. However, pattern registration will override the policy, registering..."
+            	set -x
+           	 	hzn register -p "$2" -m "${NODE_NAME}" -o "$HZN_ORG_ID" -u "$HZN_EXCHANGE_USER_AUTH" -n "$HZN_EXCHANGE_NODE_AUTH" --policy "$3"
+            	set +x
+        	fi
         fi
     fi
 
-    echo `now` "registration() end"
+    log_debug "registration() end"
 }
 
 function check_empty() {
+	log_debug "check_empty() begin"
+
     if [ -z "$1" ]; then
-        echo `now` "The ${2} value is empty, exiting..."
+        log_notify "The ${2} value is empty, exiting..."
         exit 1
     fi
+
+    log_debug "check_empty() end"
 }
 
 # checks if file or directory exists
 function check_exist() {
+	log_debug "check_exist() begin"
+
     case $1 in
 	f) if ! [[ -f "$2" ]] ; then
-			echo `now` "${3} file ${2} doesn't exist"
+			log_notify "${3} file ${2} doesn't exist"
 		    exit 1
 		fi
 	;;
 	d) if ! [[ -d "$2" ]] ; then
-			echo `now` "${3} directory ${2} doesn't exist"
+			log_notify "${3} directory ${2} doesn't exist"
 	        exit 1
 		fi
     ;;
     w) if ! ls ${2} 1> /dev/null 2>&1 ; then
-			echo `now` "${3} files ${2} do not exist"
+			log_notify "${3} files ${2} do not exist"
 	        exit 1
 	    fi
 	;;
 	*) echo "not supported"
         exit 1
 	;;
 	esac
+
+	log_debug "check_exist() end"
+}
+
+# autocomplete support for CLI
+function add_autocomplete() {
+	log_debug "add_autocomplete() begin"
+
+	log_info "Enabling autocomplete for the CLI commands..."
+	
+	SHELL_FILE="${SHELL##*/}"
+	
+	grep -q '^source /usr/local/share/horizon/hzn_bash_autocomplete.sh' ~/.${SHELL_FILE}rc || \
```

Comment:
I don't believe this is correct with the current version ... I only have:
```
# locate hzn_bash_autocomplete.sh
/etc/bash_completion.d/hzn_bash_autocomplete.sh
/root/etc/bash_completion.d/hzn_bash_autocomplete.sh
/root/tmpdeb/etc/bash_completion.d/hzn_bash_autocomplete.sh
```

After installing, logging out, re-logging I get the auto-source error:
`-bash: /usr/local/share/horizon/hzn_bash_autocomplete.sh: No such file or directory`